vendor/src/github.com/endophage/gotuf/signed/verifiers.go

```diff
@@ -47,15 +53,27 @@ func RegisterVerifier(algorithm data.SigAlgorithm, v Verifier) {
 type Ed25519Verifier struct{}
 
 func (v Ed25519Verifier) Verify(key data.PublicKey, sig []byte, msg []byte) error {
+	if key.Algorithm() != data.ED25519Key {
```

Comment:
This was an issue found by the external reviewers. Attacker could change the key type and this would attempt to use an attacker chosen verify with the wrong kind of key.